File: Kotlin/프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.kt
class Solution {
    fun solution(keymap: Array<String>, targets: Array<String>): IntArray {
        var answer: IntArray = IntArray(targets.size)

        val map = mutableMapOf<Char,Int>()

        // 등장 순서의 최솟값 넣어주기
        for (i in keymap.indices){
            for (j in keymap[i].indices){

                val c = keymap[i][j] //A

                map[c] = Math.min(map.getOrDefault(c, 200), j+1)//A =
            }
        }

        for (i in targets.indices) {
            var count = 0
            var isCounted = true
            
            for (c in targets[i]) {
                if (map.containsKey(c)) {
                    count += map[c]!!
                    continue
                }
                isCounted = false
            }
            answer[i] = if (!isCounted) -1 else count
        }

        return answer
    }
}

File: C#/프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.cs
using System.Collections.Generic;
using System;

public class Solution {
    public int[] solution(string[] keymap, string[] targets) {
        int[] answer = new int[targets.Length];
        
        Dictionary<char, int> dict = new Dictionary<char, int>();
        
        //keymap에 존재하는 char를 dict에 char와 인덱스 페어로 저장
        foreach(string key in keymap)
        {
            for(int i = 1; i <= key.Length; i++)
            {
                char k = key[i - 1];
                int idx;
                //인덱스는 항상 최저값으로 유지하기
                if(dict.TryGetValue(k, out idx))
                {
                    if(idx > i) dict[k] = i;
                    continue;
                }
                dict.Add(k, i);
            }
        }
        
        for(int i = 0; i < targets.Length; i++)
        {
            string target = targets[i];
            foreach(char t in target)
            {
                int idx;
                if(dict.TryGetValue(t, out idx) == false) 
                {
                    answer[i] = -1;
                    break;
                }
                answer[i] += idx;
            }
        }
        
        return answer;
    }
}